desktop/src/contexts/DevPodContext/Pro/ProProvider.tsx

```diff
@@ -77,11 +83,20 @@ export function ProProvider({ host, children }: { host: string; children: ReactN
       }, 1_000)
     })
 
-    return () => {
+    const canceler = () => {
       canceled = true
-      toCancel()
+      setCancelWatch(undefined)
+      setWaitingForCancel(true)
+
+      return toCancel().finally(() => setWaitingForCancel(false))
+    }
+
+    setCancelWatch({ fn: canceler })
+
+    return () => {
+      canceler()
     }
-  }, [client, store, currentProject])
+  }, [client, store, currentProject, setCancelWatch, setWaitingForCancel])
```

Comment:
you shouldn't need these here, they are guaranteed to never change

Reply:
Seems we have a different eslint config in this project. Usually, if they are missing, this is flagged.
Changed it.